### PR TITLE
Add test:no-console-mock script to disable console mocking in test-setup.js

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
Our test setup mocks console.error and console.warn so that error/warning output from jest is formatted more cleanly.  The mocking also ignores certain errors/warnings we don't care about.

A downside of this mocking is that it can make it harder to determine what the actual error is.  For instance there was an edge case in our mocking that caused another error.  This masked the original error.

After fixing that as part of this PR, I saw the original error, but also two other errors that I wasn't expecting.  I think this may just be rewrapping (unwrapping?) of the error.  This confuses other tools like wallaby which focuses on the wrong error.

In order to avoid this, I've introduced an environment variable called `GLOBAL_CONSOLE_MOCK` which when set to `false` will disable the mocking of console.warn and console.error we do in test-setup.js.  I've configured wallaby.js to always set this variable to `false`.  I've also added a script called `test:no-console-mock` to package.json so that you can see the raw output from jest when running on the command line if you want.

NOTE: it's best to review this PR with [whitespace hidden](https://github.com/Khan/perseus/pull/302/files?diff=split&w=1).

Issue: None

## Test plan:
- throw an exception in getInstructionsText() in base-radio.jsx
- do an exclusive test run of base-radio_test.jsx, see that 'Error: foo' is reported in the test file as the error

**BEFORE**
Screen Shot 2022-10-07 at 5 36 59 PM

**AFTER**
Screen Shot 2022-10-07 at 5 37 54 PM

- yarn test:no-console-mock base-radio_test.jsx, see a single error for the exception we threw

```
 FAIL  packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx
  base-radio
    edit mode
      ✕ should render <li>'s for each choice (67 ms)
      ○ skipped should toggle choice when inner element clicked
    selecting and deselecting options
      ○ skipped deselects multi-select choices
      ○ skipped select single select choices

  ● base-radio › edit mode › should render <li>'s for each choice

    foo

      348 |
      349 |     getInstructionsText: () => string = () => {
    > 350 |         throw new Error("foo");
          |               ^
      351 |         if (this.props.multipleSelect) {
      352 |             if (this.props.countChoices) {
      353 |                 return i18n._("Choose %(numCorrect)s answers:", {

      at BaseRadio.getInstructionsText (packages/perseus/src/widgets/radio/base-radio.jsx:350:15)
      at BaseRadio.getInstructionsText [as render] (packages/perseus/src/widgets/radio/base-radio.jsx:457:35)
      at finishClassComponent (node_modules/react-dom/cjs/react-dom.development.js:17160:31)
      at updateClassComponent (node_modules/react-dom/cjs/react-dom.development.js:17110:24)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18620:16)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:188:14)
      at HTMLUnknownElement.callTheUserObjectsOperation (node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:340:25)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:276:3)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:223:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23203:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21188:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:24373:3)
      at node_modules/react-dom/cjs/react-dom.development.js:24758:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:21903:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:24757:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:24840:10)
      at node_modules/@testing-library/react/dist/pure.js:101:25
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21856:12)
      at act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at render (node_modules/@testing-library/react/dist/pure.js:97:26)
      at Object.<anonymous> (packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:23:13)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:317:13)
      at runJest (node_modules/@jest/core/build/runJest.js:407:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:338:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:190:3)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 3 skipped, 4 total
Snapshots:   0 total
Time:        2.735 s
```

- yarn test base-radio_test.jsx, see three different errors including the exception we threw

```
 FAIL  packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx
  base-radio
    edit mode
      ✕ should render <li>'s for each choice (45 ms)
      ○ skipped should toggle choice when inner element clicked
    selecting and deselecting options
      ○ skipped deselects multi-select choices
      ○ skipped select single select choices

  ● base-radio › edit mode › should render <li>'s for each choice

    foo

      348 |
      349 |     getInstructionsText: () => string = () => {
    > 350 |         throw new Error("foo");
          |               ^
      351 |         if (this.props.multipleSelect) {
      352 |             if (this.props.countChoices) {
      353 |                 return i18n._("Choose %(numCorrect)s answers:", {

      at BaseRadio.getInstructionsText (packages/perseus/src/widgets/radio/base-radio.jsx:350:15)
      at BaseRadio.getInstructionsText [as render] (packages/perseus/src/widgets/radio/base-radio.jsx:457:35)
      at finishClassComponent (node_modules/react-dom/cjs/react-dom.development.js:17160:31)
      at updateClassComponent (node_modules/react-dom/cjs/react-dom.development.js:17110:24)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18620:16)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:188:14)
      at HTMLUnknownElement.callTheUserObjectsOperation (node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:340:25)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:276:3)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:223:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23203:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21188:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:24373:3)
      at node_modules/react-dom/cjs/react-dom.development.js:24758:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:21903:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:24757:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:24840:10)
      at node_modules/@testing-library/react/dist/pure.js:101:25
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21856:12)
      at act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at render (node_modules/@testing-library/react/dist/pure.js:97:26)
      at Object.<anonymous> (packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:23:13)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:317:13)
      at runJest (node_modules/@jest/core/build/runJest.js:407:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:338:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:190:3)

  ● base-radio › edit mode › should render <li>'s for each choice

    Unhandled console.error call.
    If you expect this call, then suppress it using jest.spyOn(console, "error").mockImplmentation(); otherwise, fix the test or code under test to no longer result in this console.error call.

    {"detail":{},"type":"unhandled exception"}

      106 |         .getState()
      107 |         .suppressedErrors.push(
    > 108 |             new Error(
          |             ^
      109 |                 `Unhandled console.${type} call.\n` +
      110 |                     "If you expect this call, then suppress it using " +
      111 |                     `jest.spyOn(console, "${type}").mockImplmentation(); ` +

      at reportUnhandledConsoleWarnAndErrors (config/test/test-setup.js:108:13)
      at console.reportUnhandledConsoleWarnAndErrors [as error] (config/test/test-setup.js:128:9)
      at reportException (node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:70:28)
      at innerInvokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:343:9)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:276:3)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:223:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23203:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21188:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:24373:3)
      at node_modules/react-dom/cjs/react-dom.development.js:24758:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:21903:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:24757:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:24840:10)
      at node_modules/@testing-library/react/dist/pure.js:101:25
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21856:12)
      at act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at render (node_modules/@testing-library/react/dist/pure.js:97:26)
      at Object.<anonymous> (packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:23:13)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:317:13)
      at runJest (node_modules/@jest/core/build/runJest.js:407:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:338:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:190:3)

  ● base-radio › edit mode › should render <li>'s for each choice

    Unhandled console.error call.
    If you expect this call, then suppress it using jest.spyOn(console, "error").mockImplmentation(); otherwise, fix the test or code under test to no longer result in this console.error call.

    The above error occurred in the <BaseRadio> component:
        in BaseRadio

    Consider adding an error boundary to your tree to customize error handling behavior.
    Visit https://fb.me/react-error-boundaries to learn more about error boundaries.

      106 |         .getState()
      107 |         .suppressedErrors.push(
    > 108 |             new Error(
          |             ^
      109 |                 `Unhandled console.${type} call.\n` +
      110 |                     "If you expect this call, then suppress it using " +
      111 |                     `jest.spyOn(console, "${type}").mockImplmentation(); ` +

      at reportUnhandledConsoleWarnAndErrors (config/test/test-setup.js:108:13)
      at console.reportUnhandledConsoleWarnAndErrors [as error] (config/test/test-setup.js:128:9)
      at logCapturedError (node_modules/react-dom/cjs/react-dom.development.js:19527:21)
      at logError (node_modules/react-dom/cjs/react-dom.development.js:19564:5)
      at update.callback (node_modules/react-dom/cjs/react-dom.development.js:20708:5)
      at callCallback (node_modules/react-dom/cjs/react-dom.development.js:12490:12)
      at commitUpdateQueue (node_modules/react-dom/cjs/react-dom.development.js:12511:9)
      at commitLifeCycles (node_modules/react-dom/cjs/react-dom.development.js:19883:11)
      at commitLayoutEffects (node_modules/react-dom/cjs/react-dom.development.js:22803:7)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:188:14)
      at HTMLUnknownElement.callTheUserObjectsOperation (node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:340:25)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:276:3)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:223:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at commitRootImpl (node_modules/react-dom/cjs/react-dom.development.js:22541:9)
      at unstable_runWithPriority (node_modules/scheduler/cjs/scheduler.development.js:653:12)
      at runWithPriority$1 (node_modules/react-dom/cjs/react-dom.development.js:11039:10)
      at commitRoot (node_modules/react-dom/cjs/react-dom.development.js:22381:3)
      at finishSyncRender (node_modules/react-dom/cjs/react-dom.development.js:21807:3)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21793:7)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21188:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:24373:3)
      at node_modules/react-dom/cjs/react-dom.development.js:24758:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:21903:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:24757:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:24840:10)
      at node_modules/@testing-library/react/dist/pure.js:101:25
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21856:12)
      at act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at render (node_modules/@testing-library/react/dist/pure.js:97:26)
      at Object.<anonymous> (packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:23:13)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:317:13)
      at runJest (node_modules/@jest/core/build/runJest.js:407:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:338:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:190:3)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 3 skipped, 4 total
Snapshots:   0 total
Time:        2.562 s, estimated 3 s
```

### CODE REVIEW

File: config/test/test-setup.js [hunk 1]
```diff
@@ -90,6 +90,9 @@ const reportUnhandledConsoleWarnAndErrors = (type, message, ...args) => {
         // This helper performs argument substitution to mimic what console.log
         // would do.
         let count = 0;
+        if (typeof message !== "string") {
+            message = JSON.stringify(message);
+        }
```
Comment (kevinbarabash) [inline config/test/test-setup.js:95]: Without this, I was seeing the following error which was masking the exception I was throwing as a test:
```
  ● base-radio › selecting and deselecting options › select single select choices

    TypeError: message.replace is not a function

      94 |         //     message = JSON.stringify(message);
      95 |         // }
    > 96 |         return message.replace(/%s/g, () => {
         |                        ^
      97 |             const substitution = args[count++];
      98 |             if (typeof substitution === "string") {
      99 |                 return substitution;

      at replace (config/test/test-setup.js:96:24)
      at replaceStr (config/test/test-setup.js:104:30)
      at console.reportUnhandledConsoleWarnAndErrors [as error] (config/test/test-setup.js:128:9)
      at reportException (node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:70:28)
      at innerInvokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:343:9)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:276:3)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:223:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23203:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21188:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:24373:3)
      at node_modules/react-dom/cjs/react-dom.development.js:24758:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:21903:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:24757:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:24840:10)
      at node_modules/@testing-library/react/dist/pure.js:101:25
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21856:12)
      at act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at render (node_modules/@testing-library/react/dist/pure.js:97:26)
      at Object.<anonymous> (packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:141:13)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:317:13)
      at runJest (node_modules/@jest/core/build/runJest.js:407:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:338:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:190:3)
```

File: config/test/test-setup.js [hunk 2]
```diff
@@ -120,57 +123,61 @@ const reportUnhandledConsoleWarnAndErrors = (type, message, ...args) => {
...
-
-            // react-motion
-            "Motion",
-
-            // wonder-blocks-modal-v1
-            "ScrollDisabler",
-        ];
-
-        return !components.every((name) => ignoredComponents.includes(name));
+if (process.env["GLOBAL_CONSOLE_MOCK"] !== "false") {
```
Comment (jeresig) [inline config/test/test-setup.js:126]:
```suggestion
if (process.env.GLOBAL_CONSOLE_MOCK !== "false") {
```

File: package.json [hunk 3]
```diff
@@ -114,6 +114,7 @@
     "publish:ci": "node utils/pre-publish-check-ci.js && git diff --stat --exit-code HEAD && yarn build && yarn extract-strings && changeset publish",
     "sloc": "sloc packages --exclude node_modules",
     "test": "yarn jest",
+    "test:raw": "cross-env GLOBAL_CONSOLE_MOCK=false yarn jest",
```
Comment (jeresig) [inline package.json:117]: Maybe `test:no-console-mock`? Makes the intent clearer.

**Review by jeresig — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (kevinbarabash) [reply package.json:117]: I had that originally, but it feel like a lot to type.  I'll switch it.

**Pull request merged**